scripts/newinstall.sh

```diff
@@ -557,47 +577,56 @@ git_check() {
 python_check() {
 	# Check the version by running a small Python program (taken from the Python
 	# EUPS package) XXX this will break if python is not in $PATH
-	local pyverok
-	pyverok=$(python -c 'import sys
-minver2=7
-minver3=5
-vmaj = sys.version_info[0]
-vmin = sys.version_info[1]
-if (vmaj == 2 and vmin >= minver2) or (vmaj == 3 and vmin >= minver3):
-    print(1)
-else:
-    print(0)')
-	if [[ $BATCH_FLAG = true ]]; then
-		WITH_MINICONDA=true
-	else
-		if [[ $pyverok != 1 ]]; then
+	local py_interp=python
+	local minver2=7
+	local minver3=5
+
+	local has_python=false
+	local pyverok=false
+	if n8l::has_cmd $py_interp; then
+		has_python=true
+		if n8l::pyverok $py_interp; then
```

Comment:
Yes, that is sloppy.  I prefer to work in a red-green-refactor process.  I've been looking at shell testing frameworks but they seem to be largely in a state of disrepair.  The most promising example I've seen is rspec based (ruby) with support for "mocking" shell functions.